Jenkinsfile

```diff
@@ -30,17 +30,48 @@ pipeline {
         //         }
         //     }
         // }
-        stage('Go build') {
+        stage('Go build x64') {
             agent {
                 docker {
-                    image docker.build("storj-ci", "--pull https://github.com/storj/ci.git#main").id
-                    args '--user root:root --volume "/tmp/gomod":/go/pkg/mod '
+                    image 'golang:1.17.2'
+                    args "--volume /tmp/gomod:/go/pkg/mod --user root:root"
                 }
             }
             steps {
                 script {
-                    sh './build.sh'
-                    stash(name: "build", includes: "build/")
+                    // get owner UID of working directory to run commands as that user
+                    def userId = sh(script: "ls -nd . | sed -r -e 's/[^ ]+ [^ ]+ ([^ ]+) .+/\\1/g'", returnStdout: true).trim()
```

Comment:
I believe this could be slightly cleaner as `def userId = sh(script: "stat -c '%u' .", returnStdout: true).trim()`.

Reply:
Thanks that's a lot better.